tests/robot-tests/tests/libs/common.robot

```diff
@@ -946,13 +941,17 @@ user checks list has x items
     [Arguments]    ${locator}    ${count}    ${parent}=css:body
     user waits until parent contains element    ${parent}    ${locator}
     ${list}=    get child element    ${parent}    ${locator}
-    user waits until parent contains element    ${list}    css:li    count=${count}
+    # Use xpath to more precisely get child items underneath parent list.
+    # CSS selector should not be used here as it is not precise enough with parent.
```

Comment:
The problem with the current implementation is that it would get every single child `li`, which is a real problem if you have nested lists. We specifically want to get the direct children.

Unfortunately, trying this with the child selector like `css:> li` didn't seem to work. Not sure why, but it's probably some weird interaction with the parent element context. 

The safest approach was to just go with xpath as it'll precisely get the direct child `li` elements.

Reply:
Gotcha!